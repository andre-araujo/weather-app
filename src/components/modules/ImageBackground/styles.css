.container {
  width: 100vw;
  height: 100vh;
  overflow: auto;
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 20px;
  background: #333;
  position: relative;
}

.imageContainer {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  overflow: hidden;
}

.image {
  min-width: 100%;
  min-height: 100%;
  max-height: 100vw;
  min-width: 100vh;
  height: auto;
  width: auto;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translateY(-50%) translateX(-50%);
}
