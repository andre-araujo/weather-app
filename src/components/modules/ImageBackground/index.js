import React from 'react';
import { node, string } from 'prop-types';

import styles from './styles.css';

function ImageBackground({
  children,
  imageURL,
}) {
  return (
    <div className={styles.container}>
      <figure className={styles.imageContainer}>
        <img className={styles.image} src={imageURL} alt="background" />
      </figure>
      {children}
    </div>
  );
}

ImageBackground.defaultProps = {
  children: null,
  imageURL: '#',
};

ImageBackground.propTypes = {
  children: node,
  imageURL: string,
};

export default ImageBackground;
