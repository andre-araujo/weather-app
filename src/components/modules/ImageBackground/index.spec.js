import React from 'react';
import { shallow } from 'enzyme';

import ImageBackground from './index';

describe('ImageBackground component', () => {
  it('should match snapshot', () => {
    expect(shallow(<ImageBackground />)).toMatchSnapshot();
  });

  it('should match snapshot with image background', () => {
    expect(shallow(<ImageBackground imageURL="/my-image-path" />)).toMatchSnapshot();
  });

  it('should match snapshot with children', () => {
    expect(shallow(<ImageBackground>my content</ImageBackground>)).toMatchSnapshot();
  });
});
