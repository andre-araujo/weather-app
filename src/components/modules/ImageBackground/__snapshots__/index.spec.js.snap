// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageBackground component should match snapshot 1`] = `
<div>
  <figure>
    <img
      alt="background"
      src="#"
    />
  </figure>
</div>
`;

exports[`ImageBackground component should match snapshot with children 1`] = `
<div>
  <figure>
    <img
      alt="background"
      src="#"
    />
  </figure>
  my content
</div>
`;

exports[`ImageBackground component should match snapshot with image background 1`] = `
<div>
  <figure>
    <img
      alt="background"
      src="/my-image-path"
    />
  </figure>
</div>
`;
